packages/api/src/clickhouse/index.ts

```diff
@@ -1818,21 +1818,39 @@ export const getMultiSeriesChartLegacyFormat = async ({
 
   const flatData = result.data.flatMap(row => {
     if (seriesReturnType === 'column') {
-      return series.map((_, i) => {
+      return series.map((s, i) => {
+        const groupBy =
+          s.type === 'number' ? [] : 'groupBy' in s ? s.groupBy : [];
+        const attributes = groupBy.reduce((acc, g) => {
+          acc[g] = row.group[groupBy.indexOf(g)];
```

Comment:
nit: I think we can replace indexOf with the 3rd arg in the callback to grab the index faster that way